service/job/internal/src/main/java/org/eclipse/kapua/service/job/internal/JobServiceImpl.java

```diff
@@ -187,6 +189,8 @@ public void delete(KapuaId scopeId, KapuaId jobId) throws KapuaException {
 
         //
         // Do delete
+
+        JOB_ENGINE_SERVICE.cleanJobData(scopeId, jobId);
```

Comment:
This must be run on `doPrivileged()` since the permission from the `JobEngineService.cleanJobData(...)` will change to `job:delete:*:*`.